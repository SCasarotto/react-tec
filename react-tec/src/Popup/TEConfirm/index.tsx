import React from 'react'

import {
	Popup,
	Title,
	Message,
	ButtonContainer,
	LeftButton,
	RightButton,
} from './styledComponents'
import { TEPopupProps } from 'Popup/TEPopup'

export interface TEConfirmProps extends TEPopupProps {
	title?: string
	message?: string
	leftOnClick?(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void
	leftButtonTitle?: string
	rightOnClick?(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void
	rightButtonTitle?: string
}
const TEConfirm: React.FC<TEConfirmProps> = (props) => {
	const {
		title,
		message,
		leftOnClick,
		leftButtonTitle,
		rightOnClick,
		rightButtonTitle,
		children,
		visible,
		className = '',
	} = props

	return (
		<Popup visible={visible} className={`TEConfirm ${className}`}>
			<Title className="TEConfirmTitle">{title}</Title>
			<Message className="TEConfirmMessage">{message}</Message>
			{children}
			<ButtonContainer className="TEConfirmButtonoContainer">
				<LeftButton
					onClick={leftOnClick}
					className="TEConfirmButton TEConfirmLeftButton">
					{leftButtonTitle}
				</LeftButton>
				<RightButton
					onClick={rightOnClick}
					className="TEConfirmButton TEConfirmRightButton">
					{rightButtonTitle}
				</RightButton>
			</ButtonContainer>
		</Popup>
	)
}

TEConfirm.defaultProps = {
	leftButtonTitle: 'Cancel',
	rightButtonTitle: 'Confirm',
}

export default TEConfirm
