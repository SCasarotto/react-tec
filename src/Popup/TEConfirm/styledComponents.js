import styled from 'styled-components'

import TEPopup from './../TEPopup'
import TEButton from './../../Form/TEButton'

export const Popup = styled(TEPopup)`
	.TEPopupContent {
		width: 90%;
		max-width: 600px;
		margin: 100px auto;
	}
`
export const Title = styled.h4`
	position: relative;
	width: 100%;
	text-align: center;
	font-size: 24px;
	margin-bottom: 20px;
	color: ${(props) => props.theme.darkerGray};
	padding: 30px 30px 0px 30px;
`
export const Message = styled.p`
	position: relative;
	width: 100%;
	text-align: center;
	font-size: 18px;
	line-height: 1.25;
	margin-bottom: 30px;
	padding: 0px 40px;
	color: ${(props) => props.theme.darkerGray};
`
export const ButtonContainer = styled.div`
	overflow: hidden;
	display: flex;
`
export const LeftButton = styled(TEButton)`
	width: 50%;
	font-size: 18px;
	line-height: 2;
	border-top-left-radius: 0px;
	border-top-right-radius: 0px;
	border-bottom-left-radius: 20px;
	border-bottom-right-radius: 0px;
	border-top: 1px solid ${(props) => props.theme.primary};
	border-right: 1px solid ${(props) => props.theme.primary};
	border-bottom: none;
	border-left: none;

	:hover,
	:active {
		color: ${(props) => props.theme.white};
		background-color: ${(props) => props.theme.primary};
		border-top: 1px solid ${(props) => props.theme.primary};
		border-right: 1px solid ${(props) => props.theme.primary};
		border-bottom: none;
		border-left: none;
	}
`
export const RightButton = styled(TEButton)`
	width: 50%;
	font-size: 18px;
	line-height: 2;
	border-top-left-radius: 0px;
	border-top-right-radius: 0px;
	border-bottom-left-radius: 0px;
	border-bottom-right-radius: 20px;
	border-top: 1px solid ${(props) => props.theme.primary};
	border-right: none;
	border-bottom: none;
	border-left: 1px solid ${(props) => props.theme.primary};

	:hover,
	:active {
		color: ${(props) => props.theme.white};
		background-color: ${(props) => props.theme.primary};
		border-top: 1px solid ${(props) => props.theme.primary};
		border-right: none;
		border-bottom: none;
		border-left: 1px solid ${(props) => props.theme.primary};
	}
`
