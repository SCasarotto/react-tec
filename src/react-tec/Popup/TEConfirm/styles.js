import { colors } from './../../../config/styles'

export default {
    content: {
        width: '90%',
        maxWidth: 600,
        margin: '100px auto',
    },
    title: {
        position: 'relative',
        width: '100%',
        textAlign: 'center',
        fontSize: 24,
        marginBottom: 20,
        color: colors.darkGray,
        paddingLeft: 30,
        paddingTop: 30,
        paddingRight: 30,
        paddingBottom: 0,
    },
    message: {
        position: 'relative',
        width: '100%',
        textAlign: 'center',
        fontSize: 18,
        lineHeight: 1.25,
        marginBottom: 30,
        padding: '0px 40px',
        color: colors.darkGray,
    },
    buttonContainer: {
        borderBottomRightRadius: 20,
        borderBottomLeftRadius: 20,
        overflow: 'hidden',
        display: 'flex',
    },
    button: {
        position: 'relative',
        width: '50%',
        textAlign: 'center',
        fontSize: 18,
        fontWeight: 100,
        letterSpacing: 0.5,
        color: colors.primary,
        backgroundColor: colors.white,
        lineHeight: 2,
        borderTopLeftRadius: 0,
        borderTopRightRadius: 0,
        borderBottomLeftRadius: 0,
        borderBottomRightRadius: 0,
        backgroundImage: 'none',
        transition: 'border 0.2s ease-in, color 0.2s ease-in, background-color 0.2s ease-in',
    },
    leftButton: {
        borderTop: `1px solid ${colors.primary}`,
        borderRight: `1px solid ${colors.primary}`,
        borderBottom: 'none',
        borderLeft: 'none',
        ':hover': {
            color: colors.white,
            backgroundColor: colors.primary,
            borderTop: `1px solid ${colors.primary}`,
            borderRight: `1px solid ${colors.primary}`,
            borderBottom: 'none',
            borderLeft: 'none',
        },
        ':active': {
            color: colors.white,
            backgroundColor: colors.primary,
            borderTop: `1px solid ${colors.primary}`,
            borderRight: `1px solid ${colors.primary}`,
            borderBottom: 'none',
            borderLeft: 'none',
        },
    },
    rightButton: {
        borderTop: `1px solid ${colors.primary}`,
        borderRight: 'none',
        borderBottom: 'none',
        borderLeft: `1px solid ${colors.primary}`,
        ':hover': {
            color: colors.white,
            backgroundColor: colors.primary,
            borderTop: `1px solid ${colors.primary}`,
            borderRight: 'none',
            borderBottom: 'none',
            borderLeft: `1px solid ${colors.primary}`,
        },
        ':active': {
            color: colors.white,
            backgroundColor: colors.primary,
            borderTop: `1px solid ${colors.primary}`,
            borderRight: 'none',
            borderBottom: 'none',
            borderLeft: `1px solid ${colors.primary}`,
        },
    },
}
