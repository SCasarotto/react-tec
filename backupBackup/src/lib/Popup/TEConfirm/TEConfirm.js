//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import { Popup, Title, Message, ButtonContainer, LeftButton, RightButton } from './styledComponents'

const TEConfirm = (props) => {
	const {
		title,
		message,
		leftOnClick,
		leftButtonTitle,
		rightOnClick,
		rightButtonTitle,
		children,
		visible,
		className,
	} = props

	return (
		<Popup visible={visible} className={className}>
			<Title>{title}</Title>
			<Message>{message}</Message>
			{children}
			<ButtonContainer>
				<LeftButton onClick={leftOnClick}>{leftButtonTitle}</LeftButton>
				<RightButton onClick={rightOnClick}>{rightButtonTitle}</RightButton>
			</ButtonContainer>
		</Popup>
	)
}

TEConfirm.propTypes = {
	title: PropTypes.string,
	message: PropTypes.string,
	leftOnClick: PropTypes.func.isRequired,
	leftButtonTitle: PropTypes.string,
	rightOnClick: PropTypes.func.isRequired,
	rightButtonTitle: PropTypes.string,
}

TEConfirm.defaultProps = {
	leftButtonTitle: 'Cancel',
	rightButtonTitle: 'Confirm',
}

export default TEConfirm
