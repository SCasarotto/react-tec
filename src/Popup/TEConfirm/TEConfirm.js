//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import { Popup, Title, Message, ButtonContainer, LeftButton, RightButton } from './styledComponents'

const TEConfirm = (props) => {
	const {
		title,
		message,
		leftOnClick,
		leftButtonTitle,
		rightOnClick,
		rightButtonTitle,
		children,
		visible,
		className,
	} = props

	return (
		<Popup visible={visible} className={className}>
			<Title className='TEConfirmTitle'>{title}</Title>
			<Message className='TEConfirmMessage'>{message}</Message>
			{children}
			<ButtonContainer className='TEConfirmButtonoContainer'>
				<LeftButton onClick={leftOnClick} className='TEConfirmButton TEConfirmLeftButton'>
					{leftButtonTitle}
				</LeftButton>
				<RightButton
					onClick={rightOnClick}
					className='TEConfirmButton TEConfirmRightButton'
				>
					{rightButtonTitle}
				</RightButton>
			</ButtonContainer>
		</Popup>
	)
}

TEConfirm.propTypes = {
	title: PropTypes.string,
	message: PropTypes.string,
	leftOnClick: PropTypes.func.isRequired,
	leftButtonTitle: PropTypes.string,
	rightOnClick: PropTypes.func.isRequired,
	rightButtonTitle: PropTypes.string,
}

TEConfirm.defaultProps = {
	leftButtonTitle: 'Cancel',
	rightButtonTitle: 'Confirm',
}

export default TEConfirm
