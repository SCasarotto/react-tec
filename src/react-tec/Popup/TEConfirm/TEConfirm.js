//
//TE Version 0.2.0
//

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Radium from 'radium'

import styles from './styles'

import TEPopup from './../TEPopup'
import TEButton from './../../Form/TEButton'

class TEConfirm extends Component {
	render() {
		const {
			title,
			message,
			leftOnClick,
			leftButtonTitle,
			rightOnClick,
			rightButtonTitle,
			children,
			visible,
		} = this.props

		return (
			<TEPopup visible={visible} contentStyles={styles.content}>
				<div style={styles.title}>{title}</div>
				<div style={styles.message}>{message}</div>
				{children}
				<div style={styles.buttonContainer}>
					<TEButton
						onClick={leftOnClick}
						style={{ ...styles.button, ...styles.leftButton }}
					>
						{leftButtonTitle}
					</TEButton>
					<TEButton
						onClick={rightOnClick}
						style={{ ...styles.button, ...styles.rightButton }}
					>
						{rightButtonTitle}
					</TEButton>
				</div>
			</TEPopup>
		)
	}
}

TEConfirm.propTypes = {
	title: PropTypes.string,
	message: PropTypes.string,
	leftOnClick: PropTypes.func.isRequired,
	leftButtonTitle: PropTypes.string,
	rightOnClick: PropTypes.func.isRequired,
	rightButtonTitle: PropTypes.string,
}

TEConfirm.defaultProps = {
	leftButtonTitle: 'Cancel',
}

export default Radium(TEConfirm)
