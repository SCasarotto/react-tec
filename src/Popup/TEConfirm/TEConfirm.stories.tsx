import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEConfirm, TEConfirmProps } from './TEConfirm'

export default {
	title: 'Popup/TEConfirm',
	component: TEConfirm,
	argTypes: {},
} as Meta

const Template: Story<TEConfirmProps> = (args) => <TEConfirm {...args} />

export const Default = Template.bind({})
Default.args = {}
