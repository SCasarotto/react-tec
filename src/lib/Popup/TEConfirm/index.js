import TEConfirm from './TEConfirm';

export default TEConfirm;
