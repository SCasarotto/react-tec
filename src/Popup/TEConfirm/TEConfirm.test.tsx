import React from 'react'
import { render } from '@testing-library/react'
import { TEConfirm } from './TEConfirm'

describe('TEConfirm', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(<TEConfirm visible />)
		expect(asFragment()).toMatchSnapshot()
	})
})
