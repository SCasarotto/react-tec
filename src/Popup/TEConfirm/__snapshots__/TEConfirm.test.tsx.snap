// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEConfirm matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx kmiImc TEPopup sc-hKFyIo eFASRH TEConfirm "
  >
    <div
      class="sc-gtssRu jqgoxm TEPopupContent"
    >
      <div
        class="sc-gKAblj eHPrsV TEConfirmButtonoContainer"
      >
        <button
          class="sc-dlnjPT MgKjE TEButton sc-iCoHVE gleLGG TEConfirmButton TEConfirmLeftButton"
          type="button"
        >
          Cancel
        </button>
        <button
          class="sc-dlnjPT MgKjE TEButton sc-fujyUd fBnByp TEConfirmButton TEConfirmRightButton"
          type="button"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
